src/api/space.ts

```diff
@@ -0,0 +1,59 @@
+// NOTE: use this space.ts for new API while moving the still useful ones from venue.ts to this place
+
+import Bugsnag from "@bugsnag/js";
+import firebase from "firebase/app";
+
+import { COLLECTION_SPACES, FIELD_SLUG, FIELD_WORLD_ID } from "settings";
+
+import { findWorldBySlug } from "api/world";
+
+export type FindSpaceBySlugAsIdOptions = {
+  spaceSlug: string;
+  worldId?: string;
+  worldSlug?: string;
+};
+
+export const findSpaceBySlug = async ({
+  spaceSlug,
+  worldId,
+  worldSlug,
+}: FindSpaceBySlugAsIdOptions) => {
+  if (!spaceSlug) {
+    throw new Error("The spaceSlug should be provided");
+  }
+
+  let query: firebase.firestore.Query = firebase
+    .firestore()
+    .collection(COLLECTION_SPACES)
+    .where(FIELD_SLUG, "==", spaceSlug);
+
+  if (worldId) {
+    query = query.where(FIELD_WORLD_ID, "==", worldId);
+  } else if (worldSlug) {
```

Comment:
Can you move the statement inside `else if` outside below. It will improve the readability.

Reply:
If both `worldId` and `worldSlug` are provided, only `worldId` is used and if neither is, error is thrown.

How can that move be done by keeping the same logic and improving readability?
I originally had fail fast checks above, didn't add much to readability and TS complained.

Reply:
How does 5fe64af99f91b6d7667cf7f18b0b390899632553 look?